ecs-cli/modules/cli/compose/entity/service/service_helper.go

```diff
@@ -15,22 +15,76 @@ package service
 
 import (
 	"fmt"
+	"sort"
+	"time"
 
 	log "github.com/Sirupsen/logrus"
-	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/utils/waiters"
+	ecscli "github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands"
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
 )
 
+// UpdateServiceTimeout is the time that the CLI will wait to check if the
+// count of running tasks is changing. If it has not changed then an error is thrown
+// after UpdateServiceTimeout minutes
+const DefaultUpdateServiceTimeout = 5
+const latestEventWindow = 5
+
+// serviceEvents is a wrapper for []*ecs.ServiceEvent
+// that allows us to sort it by the time stamp
+type serviceEvents []*ecs.ServiceEvent
+
+func (s serviceEvents) Len() int {
+	return len(s)
+}
+func (s serviceEvents) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+func (s serviceEvents) Less(i, j int) bool {
+	time1 := *s[i].CreatedAt
+	time2 := *s[j].CreatedAt
+	return time1.Before(time2)
+}
+
+// logNewServiceEvents logs events that have not been logged yet
+func logNewServiceEvents(loggedEvents map[string]bool, events []*ecs.ServiceEvent, cmdTimestamp time.Time) {
+
+	// sort the events so that newer ones are printed last
+	sort.Sort(serviceEvents(events))
+	for _, event := range events {
+		if _, ok := loggedEvents[*event.Id]; !ok {
+			// New event that has not been logged yet
+			loggedEvents[*event.Id] = true
+			if cmdTimestamp.Sub(*event.CreatedAt).Seconds() < latestEventWindow {
+				log.WithFields(log.Fields{
+					"timestamp": *event.CreatedAt},
+				).Info(event.Message)
+			}
+		}
+	}
+
+}
+
 // waitForServiceTasks continuously polls ECS (by calling describeService) and waits for service to get stable
 // with desiredCount == runningCount
 func waitForServiceTasks(service *Service, ecsServiceName string) error {
-	timeoutMessage := fmt.Sprintf("Timeout waiting for service %s to get stable", ecsServiceName)
+	eventsLogged := make(map[string]bool)
+	var lastRunningCount int64
+	lastRunningCountChangedAt := time.Now()
+	timeOut := float64(DefaultUpdateServiceTimeout)
+	cmdTimestamp := time.Now()
+
+	if val := service.Context().CLIContext.Float64(ecscli.ComposeServiceTimeOutFlag); val > 0 {
+		timeOut = val
+	} else if val <= 0 {
+		return fmt.Errorf("Error with timeout flag: %d is not a valid timeout value", val)
+	}
 
-	return waiters.WaitUntilComplete(func(retryCount int) (bool, error) {
+	for time.Since(lastRunningCountChangedAt).Minutes() < timeOut {
 
 		ecsService, err := service.describeService()
```

Comment:
We should wait for 15 seconds before calling the API again, otherwise, we will get throttled.